src/app/complex/complex-results/complex-navigator/complex-navigator.component.ts

```diff
@@ -30,9 +30,8 @@ export class ComplexNavigatorComponent {
   navigatorComponents: INavigatorComponent[] = [];
 
   constructor(private complexPortalService: ComplexPortalService, public state: NavigatorStateService) {
-    effect(() => {
-      this.setNavigatorComponents(this.navigatorComponentsGroupedByOrthologs(), this.navigatorComponentsWithoutGrouping());
-    });
+    effect(() => this.state.numberOfColumns.set(this.complexSearch().elements.length), {allowSignalWrites: true});
```

Comment:
Why not using computed instead of an effect for this?

Reply:
How can we use computed? `numberOfColumns` is in the `NavigatorStateService` so we need to call `this.state.numberOfColumns.set`. I don't know how to use computed to do that.